app/models/submission.rb

```diff
@@ -116,6 +119,24 @@ def evaluations_missing_or_incomplete?
     !eligible_for_evaluation? || !all_evaluations_completed? || evaluator_submission_assignments.empty?
   end
 
+  def update_submission_evaluation_status
+    calculate_and_save_evaluation_status
+  end
+
+  def calculate_and_save_evaluation_status
+    return if @updating_evaluation_status
+
+    @updating_evaluation_status = true
+    begin
+      new_status = calculate_evaluation_status
+      if evaluation_status != new_status.to_s
+        update_columns(evaluation_status: new_status)
```

Comment:
I noticed a codeclimate complaint about this. I believe if you make this a before_save instead of after_save, you can just set the attribute to the new_status instead of using `update_columns`. that should also remove the recursive callback that saves the model again while it is being saved.
```suggestion
        self.evaluation_status = new_status
```
EDIT: see my comment on the after_save first